core/src/main/java/feign/FeignException.java

```diff
@@ -37,108 +37,61 @@
-    this.request = null;
-  }
+  private Response response;
 
   protected FeignException(int status, String message) {
     super(message);
     this.status = status;
     this.request = null;
   }
 
-  protected FeignException(int status, String message, byte[] responseBody,
-      Map<String, Collection<String>> responseHeaders) {
-    super(message);
+  protected FeignException(int status, String message, Throwable cause) {
+    super(message, cause);
     this.status = status;
-    this.responseBody = responseBody;
-    this.responseHeaders = caseInsensitiveCopyOf(responseHeaders);
     this.request = null;
   }
 
-  protected FeignException(int status, String message, Request request, Throwable cause) {
-    super(message, cause);
+  protected FeignException(int status, String message, Request request) {
+    super(message);
     this.status = status;
     this.request = checkRequestNotNull(request);
   }
 
-  protected FeignException(int status, String message, Request request, Throwable cause,
-      byte[] responseBody, Map<String, Collection<String>> responseHeaders) {
+  protected FeignException(int status, String message, Request request, Throwable cause) {
     super(message, cause);
     this.status = status;
-    this.responseBody = responseBody;
-    this.responseHeaders = caseInsensitiveCopyOf(responseHeaders);
     this.request = checkRequestNotNull(request);
   }
 
-  protected FeignException(int status, String message, Request request) {
+  protected FeignException(int status, String message, Response response) {
     super(message);
     this.status = status;
-    this.request = checkRequestNotNull(request);
+    this.response = checkResponseNotNull(response);
+    this.request = checkRequestNotNull(response.request());
   }
 
-  protected FeignException(int status, String message, Request request, byte[] responseBody,
-      Map<String, Collection<String>> responseHeaders) {
-    super(message);
+  protected FeignException(int status, String message, Response response, Throwable cause) {
+    super(message, cause);
     this.status = status;
-    this.responseBody = responseBody;
-    this.responseHeaders = caseInsensitiveCopyOf(responseHeaders);
-    this.request = checkRequestNotNull(request);
+    this.response = checkResponseNotNull(response);
+    this.request = checkRequestNotNull(response.request());
   }
 
   private Request checkRequestNotNull(Request request) {
     return checkNotNull(request, EXCEPTION_MESSAGE_TEMPLATE_NULL_REQUEST);
   }
 
-  public int status() {
-    return this.status;
-  }
-
-  /**
-   * The Response Body, if present.
-   *
-   * @return the body of the response.
-   * @deprecated use {@link #responseBody()} instead.
-   */
-  @Deprecated
-  public byte[] content() {
-    return this.responseBody;
-  }
-
-  /**
-   * The Response body.
-   *
-   * @return an Optional wrapping the response body.
-   */
-  public Optional<ByteBuffer> responseBody() {
```

Comment:
as with the original constructors... we need to keep the old methods for compatibility.

Please bring them all back, not sure if they got deleted or just sorted, keeping them on their original position makes this more reviewable.

Reply:
done in the commit 2523ef1